lib/DynamoRequest.js

```diff
@@ -95,32 +80,52 @@ DynamoRequest.prototype.setReturnValues = function (returnValues) {
   return this
 }
 
-DynamoRequest.prototype.setFilter = function (filter) {
-  if (filter) {
-    this.data.ScanFilter = {}
-    for (var key in filter) {
-      var op = filter[key][0]
-      var attrList = null
-      if (op === 'BETWEEN') {
-        attrList = [typeUtil.valueToObject(filter[key][1]), typeUtil.valueToObject(filter[key][2])]
-      } else if (op === 'IN') {
-        attrList = []
-        for (var i = 1; i < filter[key].length; i += 1) {
-          attrList.push(typeUtil.valueToObject(filter[key][i]))
-        }
-      } else if (op === 'NULL' || op === 'NOT_NULL') {
-        attrList = null
-      } else {
-        // All other ops have 1 argument.
-        attrList = [typeUtil.valueToObject(filter[key][1])]
-      }
+/**
+ * @param {Array.<FilterBuilder>} conditions An array of conditions, possibly null to indicate
+ *     no conditions.
+ * @return {DynamoRequest}
+ */
+DynamoRequest.prototype.setKeyConditions = function (conditions) {
```

Comment:
These three functions, `setKeyConditions`, `setQueryFilter`, and `setScanFilter` have most code duplicated. They could be refactorized.